chart.js

```diff
@@ -0,0 +1,94 @@
+var allProductChoices = JSON.parse(localStorage.allProductChoices);
+console.log(allProductChoices);
+var cumulativeProductChoices = [];
+console.log(cumulativeProductChoices);
+var clickDataTotal = [];
+
+if (localStorage.cumulativeProductChoices) {
+  cumulativeProductChoices = JSON.parse(localStorage.cumulativeProductChoices);
+} else {
+  localStorage.cumulativeProductChoices = JSON.stringify(allProductChoices);
+  cumulativeProductChoices = allProductChoices;
+};
+
+function saveCumulativeToLocalStorage(){
+  localStorage.cumulativeProductChoices = JSON.stringify(cumulativeProductChoices);
+  console.log('Cumulative product choices have been updated in local storage!');
+}
+
+function updateCumulative(products){
+  console.log(cumulativeProductChoices.length);
+  cumulativeProductChoices = products;
+  console.log(cumulativeProductChoices.length);
+  localStorage.cumulativeProductChoices = JSON.stringify(cumulativeProductChoices);
+}
+
+function addToTotals(products){
```

Comment:
You have three functions that all run through the same loop and do slightly different things. Modularizing things is great, but to some extent it also makes sense to save time and just do them in the same function.